### ISSUE TITLE
Unable to install Windows 3.11 in Dosemu2

### ISSUE THREAD

**Issue by murkyness** [dosemu2/fdpp#174]
**Describe the bug**
Windows 3.11 Setup does not complete

**To Reproduce**
Steps to reproduce the behaviour:

1. Extract all Windows 3.11 Floppies to ~/.dosemu2/c_drive/INSTALL
2. Run Dosemu2 (no custom config)
3. CD INSTALL
4. SETUP

**Attach program/game binaries or provide an URL**

Windows 3.11 floppies can be downloaded from https://winworldpc.com/product/windows-3/311

**Additional info**

No customizations, brand new dosemu2 install.

**boot.log**
[boot.log.txt](https://github.com/dosemu2/dosemu2/files/6282350/boot.log.txt)

**Comment by stsp** [comment]
OK, as you wish.
Lets move to fdpp.
Maybe Andrew can take a
look into these FS problems?
Seems like not too many ppl
contribute to fdpp. :)

**Comment by andrewbird** [comment]
I'm not sure there is any FS problem, I suggested that as I know some ms installers check that the target is not a network drive.

**Comment by stsp** [comment]
There is likely an FS problem (in fdpp)
because @jschwartzenberg says on
other DOS things work even with mfs.

**Comment by stsp** [comment]
Hmm, @jschwartzenberg am I right
that the different mouse driver gets
you further, in fdpp??
Sounds weird.

**Comment by jschwartzenberg** [comment]
> Hmm, @jschwartzenberg am I right
> that the different mouse driver gets
> you further, in fdpp??
> Sounds weird.

Yep, I was also rather surprised to observe that. I just tested properly and it seems I was too quick. The original issue is a regression with the mouse driver, that one occurs on MS-DOS 6 too. When trying the workaround MS-DOS 6 is even more broken than FDPP unfortunately.

So we have three issues:
- default Windows mouse driver prevents the setup from continuing
- FDPP disk issue
- MS-DOS 6 crash

I guess @murkyness ran into the mouse driver issue, at least that one can be worked around. Then it's up to whether FDPP or MS-DOS 6 is easier how the reporter could proceed next.

**Comment by murkyness** [comment]
Thank you both, yes installing it on another platform and copy-pasting the working installation to .dosemu/c_drive will create a working dosemu2 windows installation.

**Comment by stsp** [comment]
What feature?

**Comment by jschwartzenberg** [comment]
I think that message is shown when trying to run a DOS application at least. Maybe there are also other circumstances.

One of the things that does work in 386 enhanced mode that does not in 286 protected mode is Video for Windows.

**Comment by murkyness** [comment]
> What feature?

The MS-DOS prompt. In 386 Enhanced mode it is supposed to be windowed instead of fullscreen.

> One of the things that does work in 386 enhanced mode that does not in 286 protected mode is Video for Windows.

Yes this doesn't work either.

**Comment by jschwartzenberg** [comment]
Cool shell :)

As I mentioned MS-DOS is known not to work. Video for Windows appears to be a regression or maybe it's related to FDPP?

**Comment by murkyness** [comment]
> Cool shell :)

:grin: 

> As I mentioned MS-DOS prompt (or any program) is known not to work. 

Ah, didn't see that

> Video for Windows appears to be a regression or maybe it's related to FDPP?

I got the exact same error (insufficient memory or disk space) when trying to install Win32s and WinG as well.

The Powerpoint viewer 97 setup error provides some more details:

![image](https://user-images.githubusercontent.com/70567689/114459403-bb54ce00-9b9d-11eb-8b68-d1227b7bfec0.png)

Microsoft Internet Explorer 5.01:

![image](https://user-images.githubusercontent.com/70567689/114459938-649bc400-9b9e-11eb-8b57-c4e477822258.png)

The Winplay 3.23b5 installer returns a general protection fault:

![image](https://user-images.githubusercontent.com/70567689/114459430-c60f6300-9b9d-11eb-97f9-34460cb69d03.png)

Same for Adobe reader:

![image](https://user-images.githubusercontent.com/70567689/114461421-1509c800-9b9f-11eb-8110-17ca20e97769.png)

And for Windows Media Player 5 beta 2:

![image](https://user-images.githubusercontent.com/70567689/114461859-80ec3080-9b9f-11eb-8d05-537c9521ae7d.png)

The Winrar, VBasic 1, and VBasic 2 installers just flash the screen without displaying any info.


Let me know if you would like me to open separate tickets for any of these.

**Comment by stsp** [comment]
No, win32s stuff is known not
to work too, sorry.
Netscape Navigator is known to
work instead of IE.
But if you are going to explore
the 32bit capabilities of win31,
then you can instead use wine. :)

**Comment by stsp** [comment]
For internet please use Trumpet
Winsock - that setup is known to work.
Or at least was...

**Comment by jschwartzenberg** [comment]
I think the disk space messages might be related to FDPP and similar to the one I get in the installer. Switching to MS-DOS (or possibly another DOS) may allow those programs to install. I could play Civilization II with its videos in DOSEMU2 some time back. The WinG and Video for Windows installers both worked without issues in the past in DOSEMU2 on MS-DOS.

Trumpet Winsock and Ultra TwinSock are both known to have worked for networking with Netscape Navigator version 2.0.

Win32s support is known to be incomplete, This is due to missing implementations in this file and maybe due to more aspects that I'm not familiar with:
https://github.com/dosemu2/dosemu2/blob/devel/src/dosext/dpmi/vxd.c

At this point it is difficult for new users to start with DOSEMU2 because you need to be aware of quite a lot of history to know what is expected to work and what is not. We're now observing both regressions and known limitations as @murkyness is trying things out. It is tricky to understand how it makes sense for people that don't know DOSEMU internals because how it behaves differs a lot from other DOS environments. In some respects it is closer to NTVDM, maybe even win9x DOS support, in others it's closer to QEMU (on KVM), etc.

We need more scripting to automate software installs and a continuous integration set-up running those to catch regressions at least. Maybe there could also be a good technical write-up of how DOSEMU works under the hood and how it compares to other environments and what that implies regarding software support. For instance that DOSEMU uses its own DPMI server. What limitations exist and which steps could be taken to overcome them and which ones cannot or are unreasonable given the overall architecture. Also what documentation is lacking to overcome part, for instance on proprietary undocumented DPMI calls used by win9x that also no other DPMI server managed to reimplement (https://github.com/sudleyplace/DPMIONE/issues/1).
